### ISSUE TITLE
Use Corretto JDK, not Zulu

### ISSUE THREAD

**Issue by iay** [ukf/ukf-members#14]
I can't recall why we're using `distribution: 'zulu'` in our CI configuration, but I will note that we're using `corretto` in `ukf-mda` so it's possible to do that. It's preferable, too, as it's the Shibboleth-supported distribution, so we should change over.

**Comment by philsmart** [comment]
I do not think the `actions/setup-java` supported corretto for Java 8

**Comment by iay** [comment]
That rings a bell! Java 17 makes a difference to a lot of things.

**Comment by philsmart** [comment]
A long time later...I believe we can, now at least, support Corretto https://github.com/actions/setup-java#supported-distributions. It should be a trivial change, so I proposed that I try this and see, so we can tick this one off?

**Comment by iay** [comment]
Yes. I don't know what the state of Zulu is these days, but Corretto is now very much what I'd suggest using for anything where you can.

**Comment by philsmart** [comment]
It also seems GitHub are going to upgrade their Node version on their runners from 20 to 24, and it is possible some of our actions are not compatible. We have three actions fixed to a version, so it seems sensible to update those too:

1. actions/checkout@v4.1.6
2. actions/setup-java@v4.2.1
3. actions/upload-artifact@v4.3.3

I am working on the branch to test this out; there might be some noise as failures happen.

**Comment by philsmart** [comment]
And our Maven build is currently broken:
```
Run mvn -B clean verify --file pom.xml
/opt/hostedtoolcache/Java_Zulu_jdk/17.0.19-10/x86/bin/java: error while loading shared libraries: libz.so.1: cannot open shared object file: No such file or directory
Error: Process completed with exit code 127.
```
Which possibly suggests our runner OS `ubuntu-latest` no longer contains libz.so. So that needs addressing as well.

**Comment by philsmart** [comment]
For the Node update, it looks like we need to update the actions to be compatible with Node 24 and then use a Runner that uses Node 24. This seems to be Runner v2.327.1 or later; see: https://github.com/actions/runner/releases/tag/v2.327.1. 

Our jobs are currently run using Runner v2.335.1 (supporting Node24). It does not appear you have any control over the runner, you get the latest every time (maybe), so we will just update the actions. (Which is something Ian has done before, looking at the commit history.)

**Comment by philsmart** [comment]
Changing to Corretto fixed our Maven build issue on the same runner. I did need to change the architecture setting from `x86` to `x64`. I do not recall why we/I used `x86` over `x64`, but Amazon certainly stopped supporting Corretto for 32-bit x86 in 2004 (with JDK 11 being the last). This was part of a larger move to remove 32-bit x86 ports from Java (JEP 503).

**Comment by philsmart** [comment]
So, to sum up, we need to fix two issues:

1. Update the actions to support Node 24
2. Update the JDK distribution to Corretto and change the architecture to x64 (we could leave it off and it would default it, but...).

**Comment by philsmart** [comment]
I updated the actions to their current latest versions (but fixed to those versions, as before). It ran successfully. 

1. setup-java: https://github.com/actions/setup-java/releases
2. checkout: https://github.com/actions/checkout/releases
3. upload-artifact: https://github.com/actions/upload-artifact/releases
